website/docs/guides/migration-guides/migrating-from-v2-to-v3.md

```diff
@@ -0,0 +1,193 @@
+        )
+)
+```
+
+The following changes need to be made when updating to KafkaFlow v3.
+
+```csharp
+.AddConsumer(
+    consumerBuilder => consumerBuilder
+        .Topic("test-topic")
+        .WithGroupId("group1")
+        .AddMiddlewares(
+            middlewares => middlewares
+                .AddDecompressor<GzipMessageDecompressor>()
+                .AddDeserializer<JsonCoreDeserializer>()                
+        )
+)
+```
+
+Not only the `.AddSerializer()` and `.AddCompressor()` were renamed to `.AddDeserializer()` and `.AddDecompressor()`, but also the concrete implementations like `JsonCoreSerializer` and `GzipMessageCompressor` were renamed to `JsonCoreDeserializer` and `GzipMessageDecompressor`. Please be aware that this change is only on the **consumer side**.
+
+Having this segregation contributes to a more consistent name scheme together with the middleware behavior. Nevertheless, other changes related to this topic were made, here is a complete list of changes related to the segregation of `ISerializer` and `ICompressor` interfaces (**consumer configuration** only):
+
+Related to Deserialization:
+
+- `IDeserializer` interface created.
+- `.AddSerializer()` renamed to `.AddDeserializer()`. 
+- Created `JsonCoreDeserializer `, `ProtobufNetDeserializer`, `ConfluentProtobufDeserializer`, `NewtonsoftJsonDeserializer`, `ConfluentJsonDeserializer` and `ConfluentAvroDeserializer` that implements the `.DeserializeAsync()` method.
+
+Related to Decompression:
+
+- `IDecompressor` interface created. 
+- `.AddCompressor()` renamed to `.AddDecompressor()`. 
+- Created `GzipMessageDecompressor` that implements the `.Decompress()` method.
+
+
+### 3.3 Consumer Batching Configuration
+
+Consumer batching configuration renamed from `.BatchConsume()` to `.AddBatching()`. Update your configuration accordingly.
+
+### 3.4. Async Support for Message Type Resolvers and Schema Registry Resolvers ([#302](https://github.com/Farfetch/kafkaflow/issues/302), [#305](https://github.com/Farfetch/kafkaflow/issues/305))
+
+In both `IMessageTypeResolvers` and `ISchemaRegistryTypeNameResolver` the interface methods only supported sync calls before KafkaFlow v3.
+To accommodate some use cases where calls to these methods can be parallelized, the methods defined in this interface were refactored to support the Async pattern.
+
+If your application has a concrete implementation of these interfaces, please make sure to update your code accordingly by using the async/await pattern. Here is the list of changes related to this subject:
+
+- `Resolve()` renamed to `ResolveAsync()` in `ISchemaRegistryTypeNameResolver` and now returns a `Task<string>`.
+- `OnConsume()` renamed to `OnConsumeAsync()` in `IMessageTypeResolver` and returns a `ValueTask<Type>`.
+- `OnProduce()` renamed to `OnProduceAsync()` in `IMessageTypeResolver` and returns a `ValueTask<Type>`.
+
+## 4. Dynamic Workers Calculation Feature
```

Comment:
Does it make sense to add the link reference for the full documentation of the dynamic workers feature?

Reply:
Added the reference like this (Please check)

## 4. Dynamic Workers Calculation Feature

One important feature added to KafkaFlow v3 was the ability to calculate how many workers are handling the message consumption dynamically. Please check the feature documentation [here](https://farfetch.github.io/kafkaflow/docs/guides/consumers/dynamic-workers-configuration).